pallets/omnipool/src/router_execution.rs

```diff
@@ -49,7 +49,8 @@ impl<T: Config> TradeExecution<OriginFor<T>, T::AccountId, T::AssetId, Balance>
 			return Ok(*state_changes.asset.delta_reserve);
 		}
 
-		let (asset_fee, protocol_fee) = T::Fee::get(&asset_out);
+		let (asset_fee, _) = T::Fee::get(&asset_out);
+		let (_, protocol_fee) = T::Fee::get(&asset_in);
```

Comment:
we are duplicating the sell function in the router? seems like a refactoring may be in order (not for this PR necessarily...)

Reply:
yeah, we know. we decided to do this way when adding router support - because we did not really want to touch and reorganize omnipool code. 

but it is time to do it now for sure.